### PR TITLE
remove next_epoch_validators, use TableVec for pending_validators

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

In order to control the size of `SuiSystemState`, in this PR we update `ValidatorSet` in the following ways:
1. remove `next_epoch_validators`
2. convert `pending_validators` as `TableVec`

As a result we also remove the validator maximal size check when calling `request_add_validator` as it is currently limited by min stake.

## Test Plan 

unit tests & local deployment

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x] user-visible impact
- [x] breaking change for a client SDKs
- [x] breaking change for FNs (FN binary must upgrade)
- [x] breaking change for validators or node operators (must upgrade binaries)
- [x] breaking change for on-chain data layout
- [x] necessitate either a data wipe or data migration

### Release notes
Remove `next_epoch_validators` from `ValidatorSet`, use `TableVec` for `pending_validators` in `ValidatorSet`

### CODE REVIEW

File: apps/wallet/src/ui/app/staking/selectors.ts [hunk 1]
```diff
@@ -49,11 +49,9 @@ export function getValidatorSelector(validatorAddress?: string) {
     return createSelector(suiSystemObjectSelector, (systemObj) => {
         const { data } = systemObj || {};
         if (data?.dataType === 'moveObject') {
-            const { active_validators: active, next_epoch_validators: next } =
-                data.fields.validators.fields;
+            const { active_validators: active } = data.fields.validators.fields;
```
Comment (longbowlu) [inline apps/wallet/src/ui/app/staking/selectors.ts:52]: @Jordan-Mysten after this change, we will need to get next epoch validators differently. We could do this by either:
1. deriving that on the client side
2. exposing an api in fullnode for this 

But the comment in line 48 seems to indicate we are looking for something different?

File: crates/sui-framework/sources/governance/sui_system.move [hunk 2]
```diff
@@ -170,10 +170,6 @@ module sui::sui_system {
         commission_rate: u64,
         ctx: &mut TxContext,
     ) {
-        assert!(
-            validator_set::next_epoch_validator_count(&self.validators) < self.parameters.max_validator_candidate_count,
```
Comment (emmazzz) [inline crates/sui-framework/sources/governance/sui_system.move:174]: Are you comfortable with us removing this check @lxfind? The upper bound is going to be auto bounded by the min stake requirement.

Comment (lxfind) [reply crates/sui-framework/sources/governance/sui_system.move:174]: I'm ok with it, but I would want to double check with the team that we are ok with the theoretical upper bound of 400 validators.

Comment (bmwill) [reply crates/sui-framework/sources/governance/sui_system.move:174]: We probably want to enforce the min stake requirement before we remove this right?

Comment (emmazzz) [reply crates/sui-framework/sources/governance/sui_system.move:174]: Fair points. We should do both then:
- ask the team if we are okay with the theoretical upper bound of 400 validators.
- keep this line here and only remove it after we complete the join validator set flow where min stake is enforced. @longbowlu

Comment (longbowlu) [reply crates/sui-framework/sources/governance/sui_system.move]: I added back the check

**Review by tnowacki — COMMENTED** [review]
Move related changes LGTM, not super familiar with the various Rust areas

**Review by emmazzz — APPROVED** [review]
lgtm

File: apps/wallet/src/ui/app/staking/selectors.ts (hunk 1 above)
Comment (longbowlu) [reply apps/wallet/src/ui/app/staking/selectors.ts:52]: was told by @Jordan-Mysten  offline that he had another change to fetch the validators in a different way

**Pull request merged**